pkgs/applications/graphics/gscreenshot/default.nix

```diff
@@ -0,0 +1,94 @@
+{ lib
+, fetchFromGitHub
+, python3Packages
+, gettext
+, gobject-introspection
+, gtk3
+, wrapGAppsHook
+, xdg-utils
+, scrot
+, slop
+, xclip
+, grim
+, slurp
+, wl-clipboard
+, waylandSupport ? true
+, x11Support ? true
+}:
+
+assert waylandSupport ->
+(grim != null && slurp != null && wl-clipboard != null);
+
+assert x11Support ->
+(scrot != null && slop != null && xclip != null && python3Packages.xlib != null);
```

Comment:
I don't think that's strictly necessary, but I'll let an actual nixpkgs committer chime in on whether to remove it.